juniper/src/integrations/url.rs

```diff
@@ -1,28 +1,34 @@
 use url::Url;
 
+use value::{ParseScalarValue, ParseScalarResult};
 use Value;
 
-graphql_scalar!(Url {
+graphql_scalar!(Url where Scalar = <S>{
     description: "Url"
 
     resolve(&self) -> Value {
-        Value::string(self.as_str())
+        Value::scalar(self.as_str().to_owned())
     }
 
     from_input_value(v: &InputValue) -> Option<Url> {
-        v.as_string_value()
+        v.as_scalar_value::<String>()
          .and_then(|s| Url::parse(s).ok())
     }
+
+    from_str<'a>(value: ScalarToken<'a>) -> ParseScalarResult<'a, S> {
+        <String as ParseScalarValue<S>>::from_str(value)
+    }
 });
 
 #[cfg(test)]
 mod test {
     use url::Url;
+    use value::DefaultScalarValue;
 
     #[test]
     fn url_from_input_value() {
         let raw = "https://example.net/";
-        let input = ::InputValue::String(raw.to_string());
+        let input: ::InputValue<DefaultScalarValue> = ::InputValue::scalar(raw.to_string());
```

Comment:
Is `InputValue<DefaultScalarValue>` needed here or does type inference handle it?

Reply:
`: InputValue` is needed, type inference could not handle this otherwise.